Add entry boxes

aabdaab opened the pull request:
This PR adds `category boxes` on top of the WebScene. It is meant as an initial screen on dataGlobe but for testing I added it to featuredGlobe (cause due to changes to layers it is easier to work on featuredGlobe for now).

Design: https://projects.invisionapp.com/d/main#/console/17484384/362411416/preview
PT task: https://www.pivotaltracker.com/story/show/165937313

This PR covers:
- add box component with title and arrow
- on box hover move box to the right (css animation)
- open a sidebar on category click
- put an active category in the url
- have a prop in url ui for boxes visibility
- initial state of entry boxes: **visible**, sidebar: **closed**
- animation on initial render (**react-spring**) and on entry box click (it's not visible now as the sidebar is not yet animated but in another task the going from entry box to sidebar will be more smooth as we can animate sidebar as well)

![entryBoxesQuality](https://user-images.githubusercontent.com/6136899/57701480-fbc70c80-7653-11e9-8d47-298549d41080.gif)

aabdaab commented:
![entryBoxesQuality](https://user-images.githubusercontent.com/6136899/57701408-dafeb700-7653-11e9-940e-c5ba9f1c1a55.gif)

kfilip94 commented:
@aabdaab Nice animation, good job here :+1: 
For now, I found two small issues, I'll take a look at the code now :)

- Add `text-shadow` property to the selected text title, like here:
e.g. `text-shadow: 0 0px 6 white;` or something similar
![Screenshot from 2019-05-14 14 47 44](https://user-images.githubusercontent.com/15097138/57703103-45652680-7657-11e9-944b-bc9b43b9ccae.png)
-  Color of `MAPPING` text should be white, not grey. I think you can also increase vertical space between `MAPPING` word and category title, according to the design.

kfilip94 commented on src/components/category-box/category-box-component.jsx:
```diff
@@ -0,0 +1,33 @@
+  </div>
+);
+
+CategoryBox.propTypes = {
```
Please add `isCategoriesBoxesVisible` to the propTypes

kfilip94 commented on src/components/category-box/category-box.js:
```diff
@@ -0,0 +1,4 @@
+// import { connect } from 'react-redux';
```
Please remove comment

kfilip94 commented:
It is probably for a different PR, I think that category boxes should disappear before sidebar animation:
![lquqr-t9une](https://user-images.githubusercontent.com/15097138/57706319-69c40180-765d-11e9-93ed-5fae1179cefc.gif)
@aabdaab have you tried to play with blur effect as a background in category boxes?

kfilip94 commented on src/pages/featured-globe/featured-globe-component.jsx:
```diff
@@ -14,23 +15,26 @@ import ZoomWidget from 'components/widgets/zoom-widget';
-    <Legend />
-  </Globe>
-);
+const FeaturedGlobeComponent = ({ sceneLayers, activeLayers, isLandscapeMode, isPaddingActive, isCategoriesBoxesVisible, handleLayerToggle, handleZoomChange }) => {
```
Don't forget to add those changes to the data globe ;)

aabdaab commented:
I don't think that blur effect applies for entry boxes, it's not reflected in the designs

weberjavi commented on src/components/entry-boxes/entry-boxes.js:
```diff
@@ -0,0 +1,22 @@
+
+const EntryBoxesContainer = props => {
+  const closeEntryBoxes = () => props.changeUI({ isCategoriesBoxesVisible: false });
+  const openSidebar = () => props.changeUI({ isPaddingActive: true });
```
What do you think of changing `view` padding when this ui state is active.
Here is a [reference on ArcGis docs](https://developers.arcgis.com/javascript/latest/api-reference/esri-views-SceneView.html#padding).
Eventually the padding would be deactivated when we enter the `fullscreen`/`clean` mode.

weberjavi commented on src/components/category-box/category-box-styles.module.scss:
```diff
@@ -0,0 +1,45 @@
+.box {
+  background-color: rgba(9, 20, 26, 0.45);
```
As @kfilip94 pointed out this should have a blurred background in case the browsers supports it.
Something like this should work:
![image](https://user-images.githubusercontent.com/6906348/57760329-9c243c00-76fb-11e9-9df0-4366d81df224.png)

weberjavi commented on src/components/entry-boxes/entry-boxes-component.jsx:
```diff
@@ -0,0 +1,50 @@
+              <CategoryBox
+                title='mapping'
+                isCategoriesBoxesVisible={isCategoriesBoxesVisible}
+                category={categories[index].name} 
```
The browser is complaining that the `type` of the prop should be an object instead of the `name` string.
![image](https://user-images.githubusercontent.com/6906348/57760490-f2917a80-76fb-11e9-9fd4-6b553e92c702.png)

weberjavi commented on src/components/category-box/category-box-component.jsx:
```diff
@@ -0,0 +1,33 @@
+    <div className={styles.categoryContainer}>
+      <p className={styles.category}>{category}</p>
+      <div className={styles.icon}>
+        <ArrowExpandIcon />
```
From the designs seems that the arrows only appear on `CategoryBox` hover.
![image](https://user-images.githubusercontent.com/6906348/57760700-66338780-76fc-11e9-8173-7bed73c5690e.png)

weberjavi commented on src/components/category-box/category-box-styles.module.scss:
```diff
@@ -0,0 +1,45 @@
+.title {
+  color: rgba(white, 0.6);
+  font-size: 10px;
+  font-family: 'PierSans';
```
Hey @aabdaab maybe we can create some `settings.scss` file to store all this variables related to fonts and colours.

aabdaab commented:
Ready again. I changed `isPaddingActive` to `isSidebarOpen` and moved `isPaddingActive` to `globe` key in the url. I didn't manage changing padding in this PR as we can leave for the full screen component to manage it as it's gonna have to reside in the `useEffect` (I tried to dynamically change the `sceneSettings` in the same matter we change our state but WebScene doesn't re-render).
Also added the same changes to the data-globe, added a blur-effect and some other small things you guys mentioned before.

kfilip94 commented on src/styles/settings.scss:
```diff
@@ -4,20 +4,24 @@ $white: #fff;
+$font-family-1: 'PierSans';
+
 // Layout 
 $sidebar-width: 280px;
```
Please use $sidebar-width variable here :)

kfilip94 commented on src/styles/ui.module.scss:
```diff
@@ -0,0 +1,22 @@
+$site-gutter: 20px;
+$blur-level: 10px;
+$sidebar-width: 280px; // use this for entry boxes and sidebar width
```
And `$sidebar-width` here :)

kfilip94 commented:
@aabdaab Thanks for changes. To me, it looks good :+1: 
One thing you could change is the space between `MAPPING` word and category title:
DESIGN:
![Screenshot from 2019-05-16 15 02 49](https://user-images.githubusercontent.com/15097138/57860072-146a2a80-77ec-11e9-86a4-8e5141796891.png)
APP:
![Screenshot from 2019-05-16 15 02 05](https://user-images.githubusercontent.com/15097138/57860082-18964800-77ec-11e9-8219-66008103da81.png)

The gap should be shorter IMO. Overall it's cool!

weberjavi commented:
Thanks @aabdaab !!
